src/p2p.rs

```diff
@@ -214,7 +215,7 @@ impl Connection {
             }
             let raw_msg = match raw_msg {
                 Ok(raw_msg) => {
-                    assert_eq!(raw_msg.magic, magic);
+                    assert_eq!(raw_msg.magic.to_hex(), magic.to_hex());
```

Comment:
Why? `ToHex` trait will be removed eventually, maybe in the upcoming release.

Reply:
`raw_msg.magic` is `[u32;4]`, `magic` is, well, `Magic`, it seemed easier to assert their hex form. I wasn't aware that `ToHex` was going to be deprecated. I'll rewrite that part.

Reply:
sorry `u32` I mean

Reply:
Pushed a new commit with `assert_eq!(Magic::from_bytes(raw_msg.magic.to_be_bytes()), magic)` instead, I reread the discussion about endianness in the rust-bitcoin PR and I think I got it right ? Alternative would be `assert_eq!(raw_msg.magic, u32::from_be_bytes(magic.to_bytes()))`, and I don't know if one is better than the other.

Reply:
You should just change the type in `RawNetworkMessage` from `u32` to `Magic` and then it should work out of the box.

Reply:
ugh, didn't figure I could do that. Done.